Core/Services/AuthenticationService.cs

```diff
@@ -88,42 +88,59 @@ public async Task<ServiceResponse> Register(RegisterRequest model)
+            var urlString = BuildUrl(token, userFromDb.UserName, _config["ReturnPaths:ConfirmEmail"]);
 
-            var uriBuilder = new UriBuilder(_config["ReturnPaths:ConfirmEmail"]);
-            var query = HttpUtility.ParseQueryString(uriBuilder.Query);
-            query["token"] = token;
-            query["username"] = userFromDb.UserName;
-            uriBuilder.Query = query.ToString();
-            var urlString = uriBuilder.ToString();
-
-            var senderEmail = _config["ReturnPaths:SenderEmail"];
-            await _emailService.SendEmailAsync(senderEmail, userFromDb.Email, "Confirm your email address", urlString);
+            await _emailService.SendEmailAsync(_config["ReturnPaths:SenderEmail"], userFromDb.Email, "Confirm your email address", urlString);
 
             return new SuccessResponse { StatusCode = HttpStatusCode.Created, Message = "User created successfully! Confirm your email." };
         }
-        public async Task<ServiceResponse> ConfirmEmail(ConfirmEmailModel model)
+
+        public async Task<ServiceResponse> ConfirmEmail(EmailDto model)
         {
             var user = await _userManager.FindByNameAsync(model.UserName);
             var result = await _userManager.ConfirmEmailAsync(user, model.Token);
 
-            if (model.UserName == null || user  == null || user.EmailConfirmed || !result.Succeeded)
+            if (model.UserName == null || user  == null || !result.Succeeded)
                 return new ErrorResponse { StatusCode = HttpStatusCode.BadRequest, Message = "Link is invalid" };
 
             return new SuccessResponse { Message = "Email confirmed succesfully" };
         }
 
-        private string CreateValidationErrorMessage(IdentityResult result)
+        public async Task<ServiceResponse> SendResetPasswordEmail(string email)
         {
-            StringBuilder builder = new StringBuilder();
-            foreach (var error in result.Errors)
-            {
-                builder.Append(error.Description + " ");
-            }
-         
-            return builder.ToString();
+            var userFromDb = await _userManager.FindByEmailAsync(email);
+            var token = await _userManager.GeneratePasswordResetTokenAsync(userFromDb);
+            var urlString = BuildUrl(token, userFromDb.UserName, _config["ReturnPaths:ResetPassword"]);
+
+            await _emailService.SendEmailAsync(_config["ReturnPaths:SenderEmail"], userFromDb.Email, "Reset your password", urlString);
+
+            return new SuccessResponse { Message = "Email to reset your password's waiting for you in mailbox" };
+        }
+
+        public async Task<ServiceResponse> ResetPassword(ResetPasswordDto model)
+        {
+            var user = await _userManager.FindByNameAsync(model.UserName);
+            var result = await _userManager.ResetPasswordAsync(user, model.Token, model.newPassword);
+
+            if (model.UserName == null || user == null || !result.Succeeded)
```

Comment:
You can shorten it to only check for result.Succeeded, as it will be false when user is null